### ISSUE TITLE
/api/v1/domain_blocks: TypeError: Cannot read property 'protocol' of undefined

### ISSUE THREAD

**Issue by Angristan** [tootsuite/mastodon#8365]
Running Mastodon 2.4.3 on mstdn.io, with no blocked/hidden domains.

Issue: 
![screenshot from 2018-08-22 14-23-21](https://user-images.githubusercontent.com/11699655/44463116-03064000-a617-11e8-9312-f1e1d11493fe.png)

"Oops! An unexpected error occurred."

In the servers logs:

```
[c2696a43-3278-441f-bdac-76a28ed4c33e] method=GET path=/api/v1/domain_blocks format=html controller=Api::V1::DomainBlocksController action=show status=200 duration=68.55 view=0.73 db=58.45
[d1736eaf-2831-4a51-a3a7-d41e506bd491] method=GET path=/api/v1/domain_blocks format=html controller=Api::V1::DomainBlocksController action=show status=200 duration=23.65 view=0.49 db=15.96
[6f939cf1-8686-429b-845e-4809748be0ed] method=GET path=/api/v1/domain_blocks format=html controller=Api::V1::DomainBlocksController action=show status=200 duration=86.71 view=0.62 db=64.36
```

Error in the console:

![screenshot from 2018-08-22 14-25-01](https://user-images.githubusercontent.com/11699655/44463231-5f695f80-a617-11e8-98f8-c181649fccef.png)

The actual XHR request is indeed a 200 but returns an empty arrays, same as other instances.

![screenshot from 2018-08-22 14-25-41](https://user-images.githubusercontent.com/11699655/44463255-71e39900-a617-11e8-9bd6-52dcaa006e7b.png)
![screenshot from 2018-08-22 14-26-11](https://user-images.githubusercontent.com/11699655/44463256-71e39900-a617-11e8-97d0-29c914c2fcd2.png)

The issue seems to be with the JS.

**Comment by Angristan** [comment]
I tried after another `assets:precompile`, without success.

**Comment by ThibG** [comment]
Fixed by #8021

**Comment by Angristan** [comment]
Awesome thanks
